### PR TITLE
feat(replay): Gap in replay timeline

### PR DESCRIPTION
**Pull request by c298lee**
Creates gaps in the timeline between `App in Background` and `App in Foreground` frames. If there's no `App in Foreground` frame after `App in Background`, the gap continues to the end of the replay.

Example gap:
image

Example gap to end of replay:
image

Relates to https://github.com/getsentry/sentry/issues/71665

### CODE REVIEW

**Review by michellewzhang — COMMENTED** [review]
generally it looks good so far! will leave the floor open for others to comment :)

File: static/app/components/replays/breadcrumbs/timelineGaps.tsx
Comment (c298lee) [reply static/app/components/replays/breadcrumbs/timelineGaps.tsx]: thanks for catching that!

File: static/app/components/replays/breadcrumbs/timelineGaps.tsx
Comment (c298lee) [reply static/app/components/replays/breadcrumbs/timelineGaps.tsx]: Updated the logic and added comments, lmk if everything makes sense!

File: static/app/components/replays/breadcrumbs/timelineGaps.tsx
Comment (c298lee) [reply static/app/components/replays/breadcrumbs/timelineGaps.tsx]: ty for looking into this!!

File: static/app/components/replays/breadcrumbs/timelineGaps.tsx
Comment (c298lee) [reply static/app/components/replays/breadcrumbs/timelineGaps.tsx:20]: I think it's used to determine the width of each column, not completely sure about why it differs tho, I copied it over from `ReplayTimelineEvents`

Comment (michellewzhang) [reply static/app/components/replays/breadcrumbs/timelineGaps.tsx:20]: ahhh cool! didn't know it was copied over, seems to work so cool w me🥳

File: static/app/components/replays/breadcrumbs/timelineGaps.tsx
Comment (michellewzhang) [inline static/app/components/replays/breadcrumbs/timelineGaps.tsx:29]:
```suggestion
  // returns all numbers in the range, exclusive of start and inclusive of stop
```

**Review by michellewzhang — APPROVED** [review]
new logic & comments look good to me - seems like we have all the cases covered. 

- no background/foreground fames --> skip the big while loop, no gaps rendered.
- only background frames for some reason --> we'll only set `start`, and it'll be a gap from `start` to the end.
- only foreground frames for some reason --> `start` never gets set, and we'll reach the end of `gapFrames`. we don't push anything to `gapCol`, so we don't render any gaps.
- mix of background/foreground frames --> iterate through the `gapFrames` matching up `start` and `end` to render the gaps, until we reach one of the cases above

(lmk if any of those sound wrong)

**Comment by michellewzhang** [comment]
would be good to test with a few more replays before merging - you can ping @romtsn or @brustolin or @krystofwoldrich to generate some!

File: static/app/components/replays/breadcrumbs/timelineGaps.tsx
Comment (michellewzhang) [inline static/app/components/replays/breadcrumbs/timelineGaps.tsx:36]: do we need to reset `currFrame` after each while loop iteration? looks like it's outside the main loop

i see we're resetting `start` and `end` to be `-1` each time we push something into `gapCol` though, which is good

File: static/app/components/replays/breadcrumbs/timelineGaps.tsx
Comment (ryan953) [inline static/app/components/replays/breadcrumbs/timelineGaps.tsx:25]: This type of filter could be put into ReplayReader and memoized. `replayTimeline.tsx` could call something like `replay.getFooFrames()` to fetch just the list we need here.

File: static/app/utils/replays/types.tsx
Comment (ryan953) [inline static/app/utils/replays/types.tsx:200]: It could be useful if we had more specific types instead of `frame is BreadcrumbFrame`

**Review by ryan953 — APPROVED** [review]
Accepting to unblock

File: static/app/utils/replays/types.tsx
Comment (c298lee) [reply static/app/utils/replays/types.tsx:200]: I get type issues when I try to use `BackgroundFrame` instead :(

Comment (ryan953) [reply static/app/utils/replays/types.tsx:200]: No. We can iterate on it later

**Comment by krystofwoldrich** [comment]
[Here](https://sentry-sdks.sentry.io/replays/7249042e9fb040cb813f7e4412824615) is an example replay with multiple app suspensions. Looks good.

**Pull request merged**
